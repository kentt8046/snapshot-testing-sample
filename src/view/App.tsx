import * as React from "react";

interface AppProps {
  text: string;
}

export const App: React.SFC<AppProps> = props => (
  <div>{props.text}</div>
);
