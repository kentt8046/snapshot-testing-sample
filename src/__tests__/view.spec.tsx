import "jest";

import * as React from "react";
import * as Enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import "jest-enzyme";

import { init } from "jest-react-vrt";

import { App } from "../view/App";

describe("view", () => {

  const { beforeAll, afterAll } = init();

  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  afterAll(() => { });

  it("snapshot", () => {
    const wrapper = Enzyme.mount(<App text="てきすと" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("screenshot", async () => {
    await expect(<App text="text" />).toMatchScreenshot();
  });

});
