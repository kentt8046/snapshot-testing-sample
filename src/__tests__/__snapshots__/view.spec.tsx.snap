// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view snapshot 1`] = `
<Component
  text="てきすと"
>
  <div>
    てきすと
  </div>
</Component>
`;
